Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -547,29 +547,28 @@ google_firestore_v1_Write Serializer::EncodeMutation(
       result.which_operation = google_firestore_v1_Write_update_tag;
       auto patch_mutation = static_cast<const PatchMutation&>(mutation);
       result.update = EncodeDocument(mutation.key(), patch_mutation.value());
-      result.update_mask = EncodeDocumentMask(patch_mutation.mask());
+      result.update_mask = EncodeFieldMask(patch_mutation.mask());
       return result;
     }
 
     case Mutation::Type::Transform: {
-      // TODO(rsgowman): Implement transform mutations. Probably like this:
-      abort();
-      /*
-        result.which_operation = google_firestore_v1_Write_transform_tag;
-        auto transform = static_cast<const TransformMutation&>(mutation);
-        result.transform.document = EncodeKey(transform.key());
-
-        size_t count = transform.field_transforms.size();
-        result.transform.field_transforms_count = count;
-        result.transform.field_transforms =
-      MakeArray<google_firestore_v1_DocumentTransform_FieldTransform>(count);
-        int i = 0;
-        for (const FieldTransform& field_transform :
-      transform.field_transforms()) { result.transform.field_transforms[i] =
-      EncodeFieldTransform(field_transform); i++;
-        }
-        return result;
-      */
+      result.which_operation = google_firestore_v1_Write_transform_tag;
+      auto transform = static_cast<const TransformMutation&>(mutation);
+      result.transform.document = EncodeString(EncodeKey(transform.key()));
```

Comment:
General question: I've seen several `Encode` functions have to be wrapped into an `EncodeString`, would it be better for them to return the nanopb-compatible type instead, so that the caller doesn't have to encode the string?

Reply:
Rich preferred this arrangement because it made it possible to use `EncodeKey` without `EncodeString` in some tests. 

My preference would be that `EncodeKey` and `EncodeQueryPath` would encode directly to the target nanopb type. Tests that want to reuse this logic could do

```c++
std::string MakeStringAndFree(pb_bytes_array_t* bytes) {
  std::string result = MakeString(bytes);
  free(bytes);
  return result;
}

std::string EncodeKey(const DocumentKey& key) {
  return MakeStringAndFree(serializer_.EncodeKey(key));
}
```

Reply:
From my (admittedly) limited exposure to `Serializer`, I would prefer encoding to the nanopb type as well. In general, I'd rather make tests more convoluted than the made code, as long as it's feasible to do.